dipy/align/transforms.pyx

```diff
@@ -0,0 +1,955 @@
+        theta : array, shape(3,)
+            buffer to write the parameters of the 2D rigid transform
+            theta[0] : rotation angle
+            theta[1] : translation along the x axis
+            theta[2] : translation along the y axis
+        """
+        theta[:3] = 0
+
+    cdef void _param_to_matrix(self, double[:] theta, double[:,:] R) nogil:
+        r""" Matrix associated with the 2D rigid transform (rotation + translation)
+
+        Parameters
+        ----------
+        theta : array, shape(3,)
+            the parameters of the 2D rigid transform
+            theta[0] : rotation angle
+            theta[1] : translation along the x axis
+            theta[2] : translation along the y axis
+        R : array, shape(3, 3)
+            buffer in which to write the rigid matrix
+        """
+        cdef:
+            double ct = cos(theta[0])
+            double st = sin(theta[0])
+        R[0,0], R[0,1], R[0,2] = ct, -st, theta[1]
+        R[1,0], R[1,1], R[1,2] = st, ct, theta[2]
+        R[2,0], R[2,1], R[2,2] = 0, 0, 1
+
+
+cdef class RigidTransform3D(Transform):
+    def __init__(self):
+        r""" Rigid transform in 3D (rotation + translation)
+        The parameter vector theta of length 6 is interpreted as follows:
+        theta[0] : rotation about the x axis
+        theta[1] : rotation about the y axis
+        theta[2] : rotation about the z axis
+        theta[3] : translation along the x axis
+        theta[4] : translation along the y axis
+        theta[5] : translation along the z axis
+        """
+        self.dim = 3
+        self.number_of_parameters = 6
+
+    cdef int _jacobian(self, double[:] theta, double[:] x, double[:,:] J) nogil:
+        r''' Jacobian matrix of a 3D rigid transform (rotation + translation)
+
+        Parameters
+        ----------
+        theta : array, shape(6,)
+            the parameters of the 3D rigid transform
+            theta[0] : rotation about the x axis
+            theta[1] : rotation about the y axis
+            theta[2] : rotation about the z axis
+            theta[3] : translation along the x axis
+            theta[4] : translation along the y axis
+            theta[5] : translation along the z axis
+        x : array, shape(3,)
+            the point at which to compute the Jacobian
+        J : array, shape(3, 6)
+            the buffer in which to write the Jacobian
+
+        Returns
+        -------
+        is_constant : int
+            always returns 0, indicating that the Jacobian is not
+            constant (it depends on the value of x)
+        '''
+        cdef:
+            double sa = sin(theta[0])
+            double ca = cos(theta[0])
+            double sb = sin(theta[1])
+            double cb = cos(theta[1])
+            double sc = sin(theta[2])
+            double cc = cos(theta[2])
+            double px = x[0], py = x[1], pz = x[2]
+
+        J[0, 0] = ( -sc * ca * sb ) * px + ( sc * sa ) * py + ( sc * ca * cb ) * pz
```

Comment:
No need for space next to the round brackets. Your call if you think it is more readable. But you could just write
 (-sc * ca * sb) * px + (sc * sa) * py + (sc * ca * cb) * pz